From 982c634accb1c9890902622a0bc82742f69909a6 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Thu, 5 Sep 2024 13:55:28 +0800
Subject: [PATCH] hostapd: Fix length of Bandwidth Indication subelement

The default length of the Bandwidth Indication subelement should be
equal to the minimum size of ieee80211_bw_ind_element structure. The
previously used value truncated this subelement by one octet.

Fixes: c7e704bdf9c3 ("hostapd: Add Bandwidth Indication subelement support for channel switch")
Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
Signed-off-by: Allen Ye <allen.ye@mediatek.com>
---
 src/ap/ieee802_11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 00fd5e4bf..35e3e710c 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -7431,7 +7431,7 @@ static u8 * hostapd_eid_bw_indication(struct hostapd_data *hapd, u8 *eid,
 {
 	u16 punct_bitmap = hostapd_get_punct_bitmap(hapd);
 	struct ieee80211_bw_ind_element *bw_ind_elem;
-	size_t elen = 3;
+	size_t elen = 4;
 
 	if (hapd->cs_freq_params.bandwidth <= 160 && !punct_bitmap)
 		return eid;
-- 
2.43.0

